pkg/types/aws_resource.go

```diff
@@ -48,4 +48,7 @@ type AWSResource interface {
 	SetStatus(AWSResource)
 	// DeepCopy will return a copy of the resource
 	DeepCopy() AWSResource
+	// PopulateResourceFromAnnotation will set the Spec or Status field that user
+	// provided from annotations
+	PopulateResourceFromAnnotation(fields map[string]string) error 
```

Comment:
Adding a new method to the `AWSResource` interface will invalidate all controllers `resource` structs that are implementing this interface. But I guess you were expecting this. 
I assume there are 2 ways for addressing it:
- PR can be merged even if PR checks associated with `iam`, `ecr`, etc controllers integration tests are failing. ACK controllers can be adjusted as a followup step when runtime version is bumped inside each controller.
- Add the `PopulateResourceFromAnnotation` method beforehand on all controllers and for all `resource` structs that are implementing the `AWSResource` interface

Reply:
Hey @adriananeci, thanks for the comment. We do have a PR pending for code-gen as well [#558](https://github.com/aws-controllers-k8s/code-generator/pull/558). We will attempt generating the controllers 
with those changes and test again.